modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/ObjectEntryOpenAPIContributor.java

```diff
@@ -438,34 +469,104 @@ else if (Objects.equals(parameterName, "relatedObjectEntryId")) {
 		return parameters;
 	}
 
+	private ObjectDefinition _getRelatedObjectDefinition(
+		ObjectRelationship objectRelationship) {
+
+		if (_objectDefinition.getObjectDefinitionId() ==
+				objectRelationship.getObjectDefinitionId2()) {
+
+			return _objectDefinitionLocalService.fetchObjectDefinition(
+				objectRelationship.getObjectDefinitionId1());
+		}
+
+		return _objectDefinitionLocalService.fetchObjectDefinition(
+			objectRelationship.getObjectDefinitionId2());
+	}
+
 	private Map<ObjectRelationship, ObjectDefinition>
-		_getRelatedObjectDefinitionsMap() {
+			_getRelatedObjectDefinitionsMap()
+		throws Exception {
 
 		Map<ObjectRelationship, ObjectDefinition> relatedObjectDefinitionsMap =
 			new HashMap<>();
 
 		List<ObjectRelationship> objectRelationships =
-			_objectRelationshipLocalService.getObjectRelationships(
+			_objectRelationshipLocalService.getAllObjectRelationships(
 				_objectDefinition.getObjectDefinitionId());
 
 		for (ObjectRelationship objectRelationship : objectRelationships) {
-			ObjectDefinition objectDefinition =
-				_objectDefinitionLocalService.fetchObjectDefinition(
-					objectRelationship.getObjectDefinitionId2());
-
 			relatedObjectDefinitionsMap.put(
-				objectRelationship, objectDefinition);
+				objectRelationship,
+				_getRelatedObjectDefinition(objectRelationship));
 		}
 
 		return relatedObjectDefinitionsMap;
 	}
 
+	private Schema _getSchema(
+		ObjectRelationship objectRelationship, String schemaName) {
+
+		ObjectSchema schema = new ObjectSchema();
+
+		String description = StringBundler.concat(
+			"Information about the relationship ", objectRelationship.getName(),
+			" can be embedded with \"nestedFields\".");
+
+		schema.setDescription(description);
+
+		schema.set$ref(schemaName);
+
+		if (((objectRelationship.getObjectDefinitionId1() ==
```

Comment:
Here you are using the following if clause:

```
if (((objectRelationship.getObjectDefinitionId1() ==
		_objectDefinition.getObjectDefinitionId()) ||
	 Objects.equals(
		 objectRelationship.getType(),
		 ObjectRelationshipConstants.TYPE_MANY_TO_MANY))
```

but in the previous method, for validating the same, you are using a different if clause:

```
if (Objects.equals(
		objectRelationship.getType(),
		ObjectRelationshipConstants.
			TYPE_MANY_TO_MANY) ||
	(Objects.equals(
		objectRelationship.getType(),
		ObjectRelationshipConstants.TYPE_ONE_TO_MANY) &&
	 (objectRelationship.getObjectDefinitionId1() ==
		 _objectDefinition.getObjectDefinitionId())))
```

Reply:
It should be consistent